### PR TITLE
[CBRD-20447] Implement a function that returns the size of a value on disk

### PR DESCRIPTION
**Pull request by funkydvd**
We need a new function in the database to show us the size of a value on disk. This is useful because we may have compressed strings and we want to see if the compression works and how much it helps.

### CODE REVIEW

**Comment by eseokoh** [comment]
@funkydvd 

one quick comment is `csql_grammar.[ch]` and `csql_lexer.[ch]` should also be pushed.
I will return to you with closer look. :)

File: src/parser/type_checking.c [hunk 1]
```diff
@@ -11944,6 +11961,10 @@ pt_upd_domain_info (PARSER_CONTEXT * parser, PT_NODE * arg1, PT_NODE * arg2, PT_
       dt->info.data_type.units = 0;
       break;
 
+    case PT_DISK_SIZE:
+      dt->info.data_type.precision = TP_FLOATING_PRECISION_VALUE;
```
Comment (ribram) [inline src/parser/type_checking.c]: shouldn't also initialize:
```c
      dt->info.data_type.dec_precision = 0;
      dt->info.data_type.units = 0;
```

File: src/parser/type_checking.c [hunk 2]
```diff
@@ -11816,7 +11833,7 @@ pt_upd_domain_info (PARSER_CONTEXT * parser, PT_NODE * arg1, PT_NODE * arg2, PT_
       || op == PT_BITSHIFT_RIGHT || op == PT_DIV || op == PT_MOD || op == PT_IF || op == PT_IFNULL || op == PT_CONCAT
       || op == PT_CONCAT_WS || op == PT_FIELD || op == PT_UNIX_TIMESTAMP || op == PT_BIT_COUNT || op == PT_REPEAT
       || op == PT_SPACE || op == PT_MD5 || op == PT_TIMEF || op == PT_AES_ENCRYPT || op == PT_AES_DECRYPT
-      || op == PT_SHA_TWO || op == PT_SHA_ONE || op == PT_TO_BASE64 || op == PT_FROM_BASE64)
+      || op == PT_SHA_TWO || op == PT_SHA_ONE || op == PT_TO_BASE64 || op == PT_FROM_BASE64 || op == PT_DISK_SIZE)
```
Comment (eseokoh) [inline src/parser/type_checking.c]: Since the return type of the function is `int`, I think there's a simpler way. 

To revert the change of line 11836 and 11958 then add `case PT_DISK_SIZE:` to 
```C
    case PT_CRC32:
      assert (dt == NULL);
      dt = pt_make_prim_data_type (parser, PT_TYPE_INTEGER);
      break;
```

Would you try it?

**Comment by eseokoh** [comment]
Thanks for the update.
It looks good. Would you merge it? :)
